src/Config.php

```diff
@@ -3415,6 +3427,42 @@ public function showFormSecurity()
         echo '</td>';
         echo '</tr>';
 
+        echo '<tr class="tab_bg_1">';
+        echo '<td colspan="4" class="center b">' . __('2FA') . '</td>';
+        echo '</tr>';
+        echo '<tr class="tab_bg_2">';
+        echo '<td>';
+        echo '<label for="dropdown_2fa_enforced' . $rand . '">';
+        echo __('Enforce 2FA');
+        echo '</label>';
+        echo '</td>';
+        echo '<td>';
+        Dropdown::showYesNo('2fa_enforced', $CFG_GLPI['2fa_enforced'], -1, ['rand' => $rand]);
+        echo '</td>';
+        echo '<td>';
+        echo '<label for="dropdown_2fa_grace_days' . $rand . '">';
+        echo __('2FA grace period (in days)');
+        echo '</label>';
+        echo '</td>';
+        echo '<td>';
+        Dropdown::showNumber('2fa_grace_days', ['value' => $CFG_GLPI["2fa_grace_days"],
+            'min'   => 0,
+            'max'   => 365,
+            'rand'  => $rand
+        ]);
+        if (!empty($CFG_GLPI['2fa_grace_date_start']) && (int) $CFG_GLPI['2fa_enforced'] === 1 && (int) $CFG_GLPI['2fa_grace_days'] > 0) {
+            $days_left = (new \Glpi\Security\TOTPManager())->getGracePeriodDaysLeft();
+            if ($days_left >= 0) {
+                $end_ts = strtotime($CFG_GLPI['2fa_grace_date_start']) + ($CFG_GLPI['2fa_grace_days'] * DAY_TIMESTAMP);
+                echo '<br><span>' . __('2FA grace period started: ') . Html::convDateTime($CFG_GLPI['2fa_grace_date_start']) . '</span>';
```

Comment:
Value position may change in the sentence depending on the locale.
```suggestion
                echo '<br><span>' . sprintf(__('2FA grace period started on %s'), Html::convDateTime($CFG_GLPI['2fa_grace_date_start'])) . '</span>';
```